library/src/test/java/com/novoda/downloadmanager/UnlinkedDataRemoverTest.java

```diff
@@ -0,0 +1,62 @@
+package com.novoda.downloadmanager;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public class UnlinkedDataRemoverTest {
+
+    private static final String LINKED_FIRST_FILE = "a filename";
+    private static final String LINKED_SECOND_FILE = "another filename";
+    private static final String UNLINKED_THIRD_FILE = "yet another filename";
+
+    private final RoomAppDatabase roomAppDatabase = mock(RoomAppDatabase.class);
+    private final LocalFilesDirectory localFilesDirectory = new FakeLocalFilesDirectory(Arrays.asList(LINKED_FIRST_FILE, LINKED_SECOND_FILE, UNLINKED_THIRD_FILE));
+    private final UnlinkedDataRemover unlinkedDataRemover = new UnlinkedDataRemover(roomAppDatabase, localFilesDirectory);
+
+    @Test
+    public void removesFiles_whenFileIsUnlinked() {
+        given(roomAppDatabase.fileNames()).willReturn(Arrays.asList(LINKED_FIRST_FILE, LINKED_SECOND_FILE));
```

Comment:
@joetimmins needed to add the `mock` when moving the `fileNames` to the `RoomAppDatabase` class. Would prefer to expose an interface on the `Database` to avoid leaking `Room` and so we can create a nice `Dummy` implementation for tests.